helion/runtime/ref_mode.py

```diff
@@ -41,27 +54,104 @@ def is_in_ref_mode_context() -> bool:
         return False
 
 
+class NoCurrentRefModeContext(RuntimeError):
+    """Raised when RefModeContext.current() is called but no context is active."""
+
+
 class RefModeContext:
     """Context manager to enable ref mode execution."""
 
-    def __init__(self, env: CompileEnvironment) -> None:
+    def __init__(self, env: CompileEnvironment, config: Config | None) -> None:
         self.env = env
         self.func_mode = RefModeTorchFunctionMode()
+        self.config = config
```

Comment:
I feel that making `CompileEnvironment` carry the ref mode config might be be great, so created the TLS above so that we can look up RefModeContext at runtime (ref mode config is for making `hl.register_tunable` work in ref mode)